notebooks/ullyses-timeseries.ipynb

```diff
@@ -0,0 +1,639 @@
+    "    \"\"\"\n",
+    "    # We will cycle through the colors in a colormap.\n",
+    "    cmap = matplotlib.colormaps[\"viridis\"]\n",
+    "    evals = np.linspace(0, 1, len(x1ds))\n",
+    "    # To keep track of legend items\n",
+    "    plot_legends = []\n",
+    "    for i,item in enumerate(x1ds):\n",
+    "        # Get ext=1 SCI data and observation date\n",
+    "        data = fits.getdata(item)\n",
+    "        date_obs = fits.getval(item, \"date-obs\", 1)\n",
+    "        time_obs = fits.getval(item, \"time-obs\", 1)\n",
+    "        color = cmap(evals[i])\n",
+    "        # Plot flux vs. wavelength\n",
+    "        for j in range(len(data[\"flux\"])): # This is to loop over the segments used.\n",
+    "            plot_legend, = plt.plot(data[\"wavelength\"][j], data[\"flux\"][j], color=color, alpha=0.6, \n",
+    "                     label=f\"{os.path.basename(item)} {date_obs} {time_obs}\")\n",
+    "        plot_legends.append(plot_legend)\n",
+    "        # Optionally set x limits\n",
+    "        if xlim is not None:\n",
+    "            plt.xlim(xlim[0], xlim[1])\n",
+    "        plt.title(\"COS/G160M spectra - CVSO-109\")\n",
+    "        plt.xlabel(\"Wavelength [$\\mathrm{\\AA}$]\")\n",
+    "        plt.ylabel(\"Flux\\n[ergs/s/cm^2/$\\mathrm{\\AA}$]\")\n",
+    "    plt.legend(handles=plot_legends)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_x1ds(x1ds)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It's a little hard to see any variability here, so let's zoom in on the C IV line."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_x1ds(x1ds, xlim=(1545, 1555))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "From this plot, we can see there seems to be two different flux states for this star, corresponding to the different observational epochs in 2014 and 2020. Because of this, the ULLYSES team cannot coadd all of the available G160M data. Instead, we create a time-series spectrum for this target."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=\"sec2\"></a>\n",
+    "## Creating a time-series spectrum"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=\"sec21\"></a>\n",
+    "### Time-series YAML files"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To create a TSS, we need to specify which grating and datasets should be used. This is done by creating a YAML file with the relevant info.\n",
+    "YAML files are convenient, human-readable files that act as configuration setups. The ULLYSES team creates timeseries YAML files for all targets that have been noted to be variable. \n",
```

Comment:
"for all targets that have been noted to be variable." -> "for all TTS targets that show variation across different observational epochs"? Or something like that to make this statement a little less strong, because I think there may be variable targets that we only have one epoch of data for, and to remove the idea that the massive stars may have tss products.